kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/ReportsCachingTests.kt

```diff
@@ -29,55 +30,80 @@ internal class ReportsCachingTests {
         addProjectWithKover {
             sourcesFrom("simple")
         }
-        reportAndCheck("SUCCESS", true)
+        reportAndCheck("SUCCESS", cached = true)
 
         run("clean", "--build-cache") {
             checkDefaultBinReport(false)
             checkDefaultReports(false)
         }
-        reportAndCheck("FROM-CACHE", true)
+        reportAndCheck("FROM-CACHE", cached = true)
     }
 
-    @SlicedGeneratedTest(allTools = true)
-    fun BuildConfigurator.testOuOfDateOnSources() {
+    @GeneratedTest(tool = CoverageToolVendor.KOVER)
+    fun BuildConfigurator.testKoverOutOfDateOnSources() {
         useLocalCache()
 
         addProjectWithKover {
             sourcesFrom("simple")
         }
-        reportAndCheck("SUCCESS", true)
+        reportAndCheck("SUCCESS", cached = true)
 
         edit("src/main/kotlin/Sources.kt") {
             "$it\n class Additional"
         }
         // tasks must be restarted after the source code is edited
-        reportAndCheck("SUCCESS", true)
+        reportAndCheck("SUCCESS", cached = true)
 
         edit("src/test/kotlin/TestClass.kt") {
             "$it\n class AdditionalTest"
         }
 
-        // tasks must be restarted after tests are edited
-        reportAndCheck("SUCCESS", true)
+        // test task must be restarted after test class is edited, but reports is up-to-date because no sources changed
```

Comment:
What happens when I edit the test class in a way that I delete some tests so the resulting coverage (of unchanged sources) can be lower? Will it trigger validation checks, then? Are you sure this is the correct behavior?

Reply:
Changing the coverage will cause the report to be regenerated.
In this case, a test class is added in which there is not a single test - therefore, this does not change the coverage